app/controllers/users_controller.rb

```diff
@@ -1,6 +1,14 @@
 class UsersController < ApplicationController
   def show
     @user = User.find(params[:id])
+    if not @user.is_instructor?
+      # List of this user's questions and answers
+      @answers = @user.answers.where.not(question_id: nil)
+      @questions = @answers.map{ |answer| answer.question }
```

Comment:
I know this can be expressed more efficiently as a join, but I haven't been able to write the join properly. @fvosburgh do you know how to write it?